code/datums/beam.dm

```diff
@@ -45,7 +45,7 @@
 			Draw()
 		sleep(sleep_time)
 
-	qdel(src)
+	PlaceInPool(src)
```

Comment:
First off, as @duncathan pointed out, this isn't the proper way to do it....secondly..

You're pooling the beam datum, but yet never calling PoolOrNew() for the beam datum anywhere; it's senseless to let these accumulate in a pool, but never get pulled back out.